File: Regression/Random Forest Regressor/Random Forest.R
# Title     : TODO
# Objective : TODO
# Created by: Chris
# Created on: 2019/07/16

dataset = read.csv("Position_Salaries.csv")
dataset = dataset[2:3]

# Create regressor
# install.packages("Rtools")
# install.packages("randomForest")
library(randomForest)
# x gives a dataframe and y gives salary.
# Thats why there are 2 different syntaxes
set.seed(123)
regressor = randomForest(x = dataset[1], y = dataset$Salary, ntree = 10)


# Create the prediction
y_pred = predict(regressor, data.frame(Level = 6.5))
print(y_pred)

# Graoh
library(ggplot2)
x_grid = seq(min(dataset$Level), max(dataset$Level), 0.01)
ggplot() +
  geom_point(aes(x = dataset$Level, y = dataset$Salary), color = "red") +
  geom_line(aes(x = x_grid, y = predict(regressor, newdata = data.frame(Level = x_grid))), color = "blue") +
  ggtitle("Yes") + 
  xlab("Level") +
  ylab("Salary")

File: Regression/Random Forest Regressor/Random Forest.py
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
from sklearn.impute import SimpleImputer
from sklearn.preprocessing import LabelEncoder, OneHotEncoder, StandardScaler
from sklearn.ensemble import RandomForestRegressor

# import the data set
dataset = pd.read_csv("Position_Salaries.csv")
x = dataset.iloc[:, 1:2].values
y = dataset.iloc[:, 2].values

regressor = RandomForestRegressor(n_estimators=10, random_state=42)
regressor.fit(x, y)

# y_pred = sc_y.inverse_transform(regressor.predict(sc_x.fit_transform([[6.5]])))
y_pred = regressor.predict([[6.5]])
print(y_pred)

x_grid = np.arange(min(x), max(x), 0.01)
x_grid = x_grid.reshape((len(x_grid), 1))

plt.scatter(x, y, color="red")
plt.plot(x_grid, regressor.predict(x_grid), color="blue")
plt.title("Hi")
plt.xlabel("Level")
plt.ylabel("Salary")
plt.show()

